kv.go

```diff
@@ -69,6 +69,9 @@ type Store interface {
 	// collected at some point after that time. Clients should not
 	// rely on the value being removed at the given time.
 	Update(ctx context.Context, key string, expire time.Time, getVal func(old []byte) ([]byte, error)) error
+
+	// Keys returns a distinct list of stored keys.
+	Keys(ctx context.Context) ([]string, error)
```

Comment:
As this extends the interface that is designed to be implemented by external things I would suggest creating a new interface that includes this one and extends it with the Keys method.

Reply:
done